pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -268,21 +276,79 @@ public void addSegment(String segmentName, TableConfig tableConfig, IndexLoading
   }
 
   public void downloadAndReplaceSegment(String segmentName, LLCRealtimeSegmentZKMetadata llcSegmentMetadata,
-      IndexLoadingConfig indexLoadingConfig) {
+      IndexLoadingConfig indexLoadingConfig, TableConfig tableConfig) {
     final String uri = llcSegmentMetadata.getDownloadUrl();
+    if (uri != null && !uri.isEmpty()) {
```

Comment:
check against uri not equal to `CommonConstants.Segments.SomeConstant` that we define as `""`. The controller and server should use the same.
Also, uri can never be null, since we never write it as null for a completed segments. You can remove that check

Reply:
Added the CommonConstants.Segments.METADATA_URI_FOR_PEER_DOWNLOAD constant.